### PR TITLE
migrate tests from integration-cli/docker_api_network_test to integration/network

### PR DESCRIPTION
**Pull request by adshmh**
**- What I did**
Migrated integration tests  from `integration-cli/docker_api_network_test.go` to `integration/network` package

**- How I did it**
Added corresponding tests under `integration/network`

**- How to verify it**

**- Description for the changelog**
Migrated tests of integration-cli/docker_api_network_test.go docker to integration/network package

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: integration-cli/docker_api_network_test.go [hunk 1]
```diff
@@ -17,15 +17,6 @@ import (
...
-func (s *DockerSuite) TestAPINetworkGetDefaults(c *check.C) {
-	testRequires(c, DaemonIsLinux)
-	// By default docker daemon creates 3 networks. check if they are present
-	defaults := []string{"bridge", "host", "none"}
-	for _, nn := range defaults {
-		c.Assert(isNetworkAvailable(c, nn), checker.Equals, true)
-	}
-}
-
 func (s *DockerSuite) TestAPINetworkCreateDelete(c *check.C) {
```
Comment (thaJeztah) [inline integration-cli/docker_api_network_test.go]: Perhaps would be good to combine a couple in a single PR; looks like there's some "easy ones" to migrate

**Comment by adshmh** [comment]
The PR has been updated. It now migrates all tests from `integration-cli/docker_api_network_test.go` to `integration/network`

**Comment by thaJeztah** [comment]
WindowsRS1 looks to be legit; https://jenkins.dockerproject.org/job/Docker-PRs-WoW-RS1/21274/console

```
21:16:30 # github.com/docker/docker/integration-cli
21:16:30 .\docker_cli_port_test.go:332:8: undefined: getNetworkResource
21:16:30 FAIL	github.com/docker/docker/integration-cli [build failed]
```



Experimental failure is a known flaky https://github.com/moby/moby/issues/37246

```
22:13:13 FAIL: docker_cli_daemon_test.go:1490: DockerDaemonSuite.TestRunContainerWithBridgeNone
```

powerpc also a flaky https://github.com/moby/moby/issues/34051

```
22:58:11 FAIL: docker_cli_swarm_test.go:1142: DockerSwarmSuite.TestSwarmLockUnlockCluster
```

File: integration/internal/network/ops.go [hunk 2]
```diff
@@ -79,15 +79,20 @@ func WithOption(key, value string) func(*types.NetworkCreate) {
 }
 
 // WithIPAM adds an IPAM with the specified Subnet and Gateway to the network
-func WithIPAM(subnet, gateway string) func(*types.NetworkCreate) {
+func WithIPAM(driver, subnet, gateway, IPRange string) func(*types.NetworkCreate) {
```
Comment (vdemeester) [inline integration/internal/network/ops.go]: We may want to make this more optional (i.e. something like…)

```go
func WithIPAM(op ...func(*network.IPAMConfig)) func(*types.NetworkCreate) {
// …
}

// Using like
network.WithIPAM(Driver("foo"), Subnet(…), …)
```

But that could be done in a follow-up :wink:

File: integration/network/inspect_test.go [hunk 3]
```diff
@@ -2,21 +2,115 @@ package network // import "github.com/docker/docker/integration/network"
...
+	"github.com/docker/docker/internal/test/request"
 	"gotest.tools/assert"
+	is "gotest.tools/assert/cmp"
 	"gotest.tools/poll"
+	"gotest.tools/skip"
 )
 
 const defaultSwarmPort = 2477
 
+func TestNetworkFilter(t *testing.T) {
```
Comment (vdemeester) [inline integration/network/inspect_test.go]: This is a weird test :sweat_smile: Shouldn't it be : get the id from `NetworkInspect`, filter the network list by the name `bridge` _and_ assert it's the same id ?

**Review by vdemeester — COMMENTED** [review]
SGTM with few comments :angel:

File: integration/internal/network/ops.go (hunk 2 above)
Comment (adshmh) [reply integration/internal/network/ops.go]: Thank you for the review. I will submit a PR for this.

File: integration/network/inspect_test.go (hunk 3 above)
Comment (adshmh) [reply integration/network/inspect_test.go]: Thank you for the review. Fixed.

**Comment by thaJeztah** [comment]
Quite some flaky tests, but this one I hadn't seen before https://jenkins.dockerproject.org/job/Docker-PRs/50107/console

```
15:28:44 --- FAIL: TestNetworkFilter (0.02s)
15:28:44 	inspect_test.go:36: assertion failed: 1 (int) != 2 (int)
```

**Comment by adshmh** [comment]
`TestNetworkFilter` is added here as a replacement for the integration test `TestAPINetworkFilter`. I don't know yet why it is flaky; thought there should only be a single network with the name 'bridge'.

**Comment by olljanat** [comment]
@adshmh this one needs to be rebased

**Comment by cpuguy83** [comment]
Closing this one since it is stale.

**Pull request closed without merging**